packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -141,9 +142,14 @@ export const useField = <
     });
   });
 
-  const handleInputBlur = useEventCallback((...args) => {
-    onBlur?.(...(args as []));
-    setSelectedSections(null);
+  const handleInputBlur = useEventCallback((event: React.FocusEvent<HTMLInputElement>, ...args) => {
+    const isClearButton: Boolean = Boolean(
+      event.relatedTarget?.className?.split(' ')?.includes('deleteIcon'),
+    );
+    if (!isClearButton) {
+      onBlur?.(event, ...(args as []));
+      setSelectedSections(null);
+    }
```

Comment:
This is probably the reason why [Autocomplete](https://mui.com/material-ui/react-autocomplete/) avoids having this clearable button in the focus sequence and sets `tabindex={-1}`.